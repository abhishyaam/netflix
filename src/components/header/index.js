import React from 'react';
import { useState } from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
import {
  Background,
  Container,
  Logo,
  ButtonLink,
  Feature,
  Text,
  FeatureCallOut,
  Link,
  Group,
  Picture,
  Dropdown,
  Profile,
  SearchIcon,
  Search,
  SearchInput,
  PlayButton,
} from './styles/header';

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture src={`/images/users/${src}.png`} {...restProps} />;
};

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSeacrhActive] = useState(false);
  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSeacrhActive((searchActive) => !searchActive)}
      >
        <img src='/images/icons/search.png' alt='Search' />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        placeholder='Search films and series'
        active={searchActive}
      />
    </Search>
  );
};
