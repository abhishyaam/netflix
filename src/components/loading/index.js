import React from 'react';
import { Picture, Spinner, LockBody, ReleaseBody } from './styles/loading';

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner {...restProps}>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} data-testid='loading-picture' />
    </Spinner>
  );
}

Loading.ReleaseBody = function LoadingRealeaseBody({ src, ...restProps }) {
  return <ReleaseBody />;
};
