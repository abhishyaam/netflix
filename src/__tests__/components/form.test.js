import React from 'react';
import { render } from '@testing-library/react';
import { Form } from '../../components';
jest.mock('react-router-dom');
describe('<Form />', () => {
  it('renders the <Form /> with populated data', () => {
    const { container, queryByText, getByPlaceholderText } = render(
      <Form>
        <Form.Title>Sign In Now</Form.Title>

        <Form.Base>
          <Form.Input placeholder='Email Address' onChange={() => {}} />
          <Form.Input
            type='password'
            placeholder='Password'
            onChange={() => {}}
          />
          <Form.Submit type='submit' disabled>
            Sign In
          </Form.Submit>
        </Form.Base>

        <Form.TextSmall>
          This page is protected by Google reCAPTCHA to ensure you're not a
          bot.Learn more.
        </Form.TextSmall>
      </Form>
    );
    expect(
      queryByText(
        "This page is protected by Google reCAPTCHA to ensure you're not a bot.Learn more."
      )
    ).toBeTruthy();
    expect(queryByText('Sign In')).toBeTruthy();
    expect(queryByText('Sign In').disabled).toBeTruthy();
    expect(getByPlaceholderText('Email Address')).toBeTruthy();
    expect(getByPlaceholderText('Password')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Form /> with populated data', () => {
    const { container, queryByText, getByPlaceholderText } = render(
      <Form>
        <Form.Error>Your Email address is already in use</Form.Error>
        <Form.Submit>Sign in</Form.Submit>
      </Form>
    );
    expect(queryByText('Your Email address is already in use')).toBeTruthy();
    expect(queryByText('Sign in').disabled).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
