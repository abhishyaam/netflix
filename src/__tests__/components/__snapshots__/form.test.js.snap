// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders the <Form /> with populated data 1`] = `
<div
  class="form__Container-sc-7952rh-0 kjWqOt"
>
  <h1
    class="form__Title-sc-7952rh-3 cudFWu"
  >
    Sign In Now
  </h1>
  <form
    class="form__Base-sc-7952rh-2 hCZaqV"
  >
    <input
      class="form__Input-sc-7952rh-7 eKjyQA"
      placeholder="Email Address"
    />
    <input
      class="form__Input-sc-7952rh-7 eKjyQA"
      placeholder="Password"
      type="password"
    />
    <button
      class="form__Submit-sc-7952rh-8 cdwoKf"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p
    class="form__TextSmall-sc-7952rh-5 epwuoT"
  >
    This page is protected by Google reCAPTCHA to ensure you're not a bot.Learn more.
  </p>
</div>
`;

exports[`<Form /> renders the <Form /> with populated data 2`] = `
<div
  class="form__Container-sc-7952rh-0 kjWqOt"
>
  <div
    class="form__Error-sc-7952rh-1 lewbJn"
  >
    Your Email address is already in use
  </div>
  <button
    class="form__Submit-sc-7952rh-8 cdwoKf"
  >
    Sign in
  </button>
</div>
`;
